sdtensor.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef UTENSOR_SDTENSOR_H
+#define UTENSOR_SDTENSOR_H
+#include "tensor.hpp"
+#include "tensorIdxImporter.hpp"
+template <class T>
+class SDTensor : public Tensor {
+  public:
+    SDTensor(TName _name, std::string filename) : Tensor(_name) {
+        _filename = filename;
+        cursor = 0;
+        dirty = false;
+    }
+
+    SDTensor(std::initializer_list<uint32_t> l, TName _name, std::string file) : Tensor(_name) {
+      std::vector<uint32_t> v;
+      for (auto i : l) {
+         v.push_back(i);
+      }
```

Comment:
With proper const usage, cpp will auto deep copy std::initializer_lists to std::vector.

Reply:
@mbartling We used initializer list because it has a cleaner syntax than vectors. It is good to learn that init list gets converted to vectors. I'd propose we merge this PR and create a new PR correcting all sub-optimal uses of init list.